spacepy/pybats/bats.py

```diff
@@ -640,40 +646,122 @@ def trace(self, bats):
+    A child class of :class:`~pybats.IdlFile` tailored to 2D BATS-R-US output.
+
+    Calculations
+    ------------
+    
+    New values can be added via the addition of new keys.  For example, 
+    a user could add radial distance to an equatorial Bats2d object as follows:
+
+    >>> import numpy as np
+    >>> from spacepy.pybats import bats
+    >>> mhd = bats.Bats2d('z=0_example.out')
+    >>> mhd['rad'] = np.sqrt( mhd['x']**2 + mhd['y']**2 )
+
+    Note, however, that if the user switches the data frame in a *.outs file
+    to access data from a different epoch, these values will need to be
+    updated.
+
+    An exception to this is built-in `calc_*` methods, which perform common
+    MHD/fluid dynamic calculations (i.e., Alfven wave speed, vorticity, and
+    more.)  These values are updated when the data frame is switched (see the
+    `switch_frame` method).
+
+    Plotting
+    --------
+
+    While users can employ Matplotlib to plot values, a set of built-in
+    methods are available to expedite plotting.  These are the
+    `add_<plot type>` methods.  These methods always have the following
+    keyword arguments that allow users to optionally build more complicated
+    plots: *target* and *loc*.  The *target* kwarg tells the plotting method
+    where to place the plot and can either be a Matplotlib figure or axes
+    object.  If it's an axes object, *loc* sets the subplot location using
+    the typical matplotlib syntax (e.g., `loc=121`).  The default behavior is
+    to create a new figure and axes object.
+
+    This approach allows a user to over-plot contours, field lines, and
+    other plot artists as well as combine different subplots onto a single
+    figure.  Continuing with our example above, let's plot the grid layout
+    for our file as well as equatorial pressure and flow streamlines:
+
+    >>> import matplotlib.pyplot as plt
+    >>> fig = plt.Figure(figsize=(8,6))
+    >>> mhd.add_grid_plot(target=fig, loc=121)
+    >>> mhd.add_contour('x','y','p', target=fig, loc=122)
+    >>> mhd.add_stream_scatter('ux', 'uy', target=fig, loc=122)
+
+    Useful plotting methods include the following:
+
+    | Plot Method        | Description                                    |
+    | ------------------ | ---------------------------------------------- |
+    | add_grid_plot      | Create a quick-look diagram of the grid layout |
+    | add_contour        | Create a contour plot of a given variable      |
+    | add_pcolor         | Add a p-color (no-interpolation contour) plot  |
+    | add_stream_scatter | Scatter stream traces (any vector field)       |
+    | add_b_magsphere    | Add magnetic field lines for X-Z plane cuts    |
+    | add_planet         | Add a simple black/white planet at the origin  |
+    | add_body           | Add an inner boundary at the origin            |
+
+    Extracting and Stream Tracing
+    -----------------------------
+
```

Comment:
Same question about whether blank lines under section headings affect the rendering in docs